### ISSUE TITLE
Implement handling of _NET_DESKTOP_VIEWPORT

### ISSUE THREAD

**Issue by blueyed** [awesomeWM/awesome#759]
https://specifications.freedesktop.org/wm-spec/wm-spec-latest.html#idm140200472713248

This would allow rofi to get the x/y coordinates for the current desktop (which is a tag in awesome):
https://specifications.freedesktop.org/wm-spec/wm-spec-latest.html#idm140200472713248

**Comment by psychon** [comment]
Uhm. What should that be set to, a screen's geometry? Also, the C code doesn't know that a tag belongs to a screen, I think. That's implemented purely in Lua.

Also, rofi doesn't seem to implement this: https://github.com/DaveDavenport/rofi/search?utf8=%E2%9C%93&q=_NET_DESKTOP_VIEWPORT

**Comment by blueyed** [comment]
Got the 2nd link wrong in the initial comment.. rofi uses this: https://github.com/DaveDavenport/rofi/blob/5d7bf1cfcf53f3f56a63ef5daf714e82c30adbbf/source/x11-helper.c#L258.

Is there another way to ask awesome about the current monitor/display and the relative x/y coordinates in the xinerama space?!

**Comment by psychon** [comment]
Hm, it uses `xcb_ewmh_get_current_desktop()`. If multiple tags are selected (e.g. one on screen 1, another one on screen 2), awesome only places the "first" tag in there (since EWMH doesn't allow multiple values). So this way to get the currently "focused" tag doesn't even work with awesome.
https://specifications.freedesktop.org/wm-spec/wm-spec-latest.html#idm140200472709456

If we really want to support `_NET_DESKTOP_VIEWPORT`, we could just make that something which can be set from Lua. However, I don't think that it would really help you. Awesome's tag concept just doesn't fit into EWMH.

**Comment by blueyed** [comment]
Thanks for your great explanation as always!

To repeat myself:

> Is there another way to ask awesome about the current monitor/display and the relative x/y coordinates in the xinerama space?!

For the moment I have a wrapper `rofi` script:

    #!/bin/sh

    monitor=$(echo 'return require("awful").screen.focused() - 1' \
      | awesome-client | sed -E 's/^[^0-9]+//')

    /usr/bin/rofi -monitor $monitor "$@"

btw: is there some pattern for "execute the script with my name, but ignore myself from $PATH"? (instead of hard-coding /usr/bin/rofi there)

**Comment by psychon** [comment]
Nope, there is no such way right know (besides using awesome-client plus some magic). If it would help you, we could add something like an `AWESOME_SELECTED_TAGS` property that works just like `_NET_CURRENT_DESKTOP`, but can have multiple values, but then the `_NET_DESKTOP_VIEWPORT`-issue is still left.

And no, I do not know any such pattern.
